src/core/structures/PDFAcroFormField.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  PDFArray,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFObject,
+  PDFString,
+} from 'src/core';
+
+const acroFormFieldTypes: Array<PDFObject | undefined> = [
+  PDFName.Btn,
+  PDFName.Ch,
+  PDFName.Tx,
+  PDFName.Sig,
+];
+
+class PDFAcroFormField {
+  static fromDict(dict: PDFDict): PDFAcroFormField {
+    const ft = PDFName.of('FT');
+    const hasValidFieldType =
+      dict.has(ft) && acroFormFieldTypes.includes(dict.get(ft));
+    if (!hasValidFieldType) {
+      throw new Error('Invalid PDFAcroFormField Type');
+    }
+    return new PDFAcroFormField(dict);
+  }
+
+  private readonly dict: PDFDict;
```

Comment:
I think we should leave these underlying values `readonly`, but publicly accessible (same goes for [`PDFAcroForm.dict`](https://github.com/Hopding/pdf-lib/blob/00c8d7706c912bce9615afc087ed7b0afca7e95b/src/core/structures/PDFAcroForm.ts#L19), actually). In a perfect world, users would never need to dip into the low-level guts and could do everything they needed via the high level APIs. But in practice, there's always a minority of users that need to get into the nitty gritty of things.

Reply:
Updated